src/services/ContributingDeveloperAuditService/scm/GitHub/api/GitHubApiClient.ts

```diff
@@ -0,0 +1,215 @@
+import axios, { AxiosInstance, AxiosResponse } from "axios";
+import { soosLogger } from "../../../../../logging/SOOSLogger";
+import { sleep } from "../../../../../utilities";
+
+interface IHttpRequestParameters {
+  baseUri: string;
+  githubPAT: string;
+}
+
+interface IHttpClientParameters extends IHttpRequestParameters {
+  apiClientName: string;
+}
+
+export interface GitHubOrganization {
+  login: string;
+}
+
+export interface GitHubRepository {
+  id: number;
+  name: string;
+  full_name: string;
+  private: boolean;
+  owner: GitHubOrganization;
+}
+
+export interface ContributingDeveloper {
+  username: string;
+  repositories: ContributingDeveloperRepositories[];
+}
+
+interface ContributingDeveloperRepositories {
+  id: number;
+  name: string;
+  lastCommit: string;
+  isPrivate: boolean;
+}
+
+export interface Commits {
+  commit: {
+    author: Author;
+  };
+}
+
+export interface Author {
+  name: string;
+  email: string;
+  date: string;
+}
+
+export const threeMonthsDate = (() => {
+  const d = new Date();
+  d.setDate(d.getDate() - 90);
+  return `${d.getUTCFullYear()}-${
+    d.getMonth() + 1
+  }-${d.getUTCDate()}T${d.getHours()}:${d.getMinutes()}:${d.getSeconds()}Z`;
+})();
+
+class GitHubApiClient {
+  private readonly client: AxiosInstance;
+
+  constructor(githubPAT: string, baseUri: string = "https://api.github.com") {
+    this.client = GitHubApiClient.createHttpClient({
+      baseUri,
+      githubPAT,
+      apiClientName: "GitHub API",
+    });
+  }
+
+  private static createHttpClient({ baseUri, githubPAT, apiClientName }: IHttpClientParameters) {
+    const client = axios.create({
+      baseURL: baseUri,
+      headers: {
+        accept: "application/vnd.github+json",
+        Authorization: `Bearer ${githubPAT}`,
+      },
+    });
+
+    client.interceptors.request.use(
```

Comment:
if we are going to repeat this code, we should make it shared - but not sure we need anything but the request url / response codes here

Reply:
Think would be okay to wait until next scm implementation and see, I can put a TODO - or NOTE in here just in case